Improve ruleSyntax option documentation (#10441)
traefik/traefik: docs/content/routing/routers/index.md

### RuleSyntax

_Optional, Default=""_

In Traefik v3 a new rule syntax has been introduced ([migration guide](../../migration/v2-to-v3.md#router-rule-matchers)).
`ruleSyntax` option allows to configure the rule syntax to be used for parsing the rule on a per-router basis.
This allows to have heterogeneous router configurations and ease migration.

> There is no default value for the `ruleSyntax` option per se, and when undefined,
> it is inherited from the `defaultRuleSyntax` option in the static configuration.

Reviewer: ```suggestion
The default value of the `ruleSyntax` option is inherited from the `defaultRuleSyntax` option in the static configuration.
```
There is a default :wink:

> By default, the `defaultRuleSyntax` static option is automatically set to `v3`, meaning that the default rule is the new one.

Reviewer: ```suggestion
By default, the `defaultRuleSyntax` static option is automatically set to `v3`, meaning that the default rule syntaxt is the new one.
```

### RuleSyntax

_Optional, Default=""_

In Traefik v3 a new rule syntax has been introduced ([migration guide](../../migration/v2-to-v3.md#router-rule-matchers)).
`ruleSyntax` option allows to configure the rule syntax to be used for parsing the rule on a per-router basis.
This allows to have heterogeneous router configurations and ease migration.

> There is no default value for the `ruleSyntax` option per se, and when undefined,
> it is inherited from the `defaultRuleSyntax` option in the static configuration.

Reviewer: ```suggestion
The default value of the `ruleSyntax` option is inherited from the `defaultRuleSyntax` option in the static configuration.
```

> By default, the `defaultRuleSyntax` static option is automatically set to `v3`, meaning that the default rule is the new one.

Reviewer: ```suggestion
By default, the `defaultRuleSyntax` static option is automatically set to `v3`, meaning that the default rule syntax is the new one.
```
